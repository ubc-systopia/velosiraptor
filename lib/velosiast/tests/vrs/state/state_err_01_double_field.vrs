state = StateDef(base: addr) {
    reg foo [ 8 ],
    reg foo [ 8 ],
};


state = StateDef(base: addr) {
    mem foo [ base, 8, 8 ],
    mem foo [ base, 8, 8 ],
};


state = StateDef(base: addr) {
    mem foo [ base, 8, 8 ],
    reg foo [ 8 ],
};

state = StateDef(base: addr) {
    reg foo [ 8 ],
    mem foo [ base, 8, 8 ],
};

state = StateDef() {
    reg foo [ 8 ] {
        0 .. 32 foo,
        32 .. 64 foo
    }
};

state = StateDef() {
    mem foo [ base, 0, 8 ] {
        0 .. 32 foo,
        32 .. 64 foo
    }
};