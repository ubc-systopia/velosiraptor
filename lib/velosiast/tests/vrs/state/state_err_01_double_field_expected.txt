error: duplicate definitions with name `state.foo`
     --> tests/vrs/state/state_err_01_double_field.vrs:3:5
      |
    2 |      reg foo [ 8 ],
      |      ^^^^^^^^^^^^^ previous definition of `state.foo` was here
   ...
      |
    3 |      reg foo [ 8 ],
      |      ^^^^^^^^^^^^^ duplicate definition
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

error: duplicate definitions with name `state.foo`
     --> tests/vrs/state/state_err_01_double_field.vrs:9:5
      |
    8 |      mem foo [ base, 8, 8 ],
      |      ^^^^^^^^^^^^^^^^^^^^^^ previous definition of `state.foo` was here
   ...
      |
    9 |      mem foo [ base, 8, 8 ],
      |      ^^^^^^^^^^^^^^^^^^^^^^ duplicate definition
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

error: duplicate definitions with name `state.foo`
     --> tests/vrs/state/state_err_01_double_field.vrs:15:5
      |
   14 |      mem foo [ base, 8, 8 ],
      |      ^^^^^^^^^^^^^^^^^^^^^^ previous definition of `state.foo` was here
   ...
      |
   15 |      reg foo [ 8 ],
      |      ^^^^^^^^^^^^^ duplicate definition
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

error: duplicate definitions with name `state.foo`
     --> tests/vrs/state/state_err_01_double_field.vrs:20:5
      |
   19 |      reg foo [ 8 ],
      |      ^^^^^^^^^^^^^ previous definition of `state.foo` was here
   ...
      |
   20 |      mem foo [ base, 8, 8 ],
      |      ^^^^^^^^^^^^^^^^^^^^^^ duplicate definition
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

error: duplicate definitions with name `state.foo.foo`
     --> tests/vrs/state/state_err_01_double_field.vrs:26:9
      |
   25 |          0 .. 32 foo,
      |          ^^^^^^^^^^^ previous definition of `state.foo.foo` was here
   ...
      |
   26 |          32 .. 64 foo
      |          ^^^^^^^^^^^^ duplicate definition
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

error: duplicate definitions with name `state.foo.foo`
     --> tests/vrs/state/state_err_01_double_field.vrs:33:9
      |
   32 |          0 .. 32 foo,
      |          ^^^^^^^^^^^ previous definition of `state.foo.foo` was here
   ...
      |
   33 |          32 .. 64 foo
      |          ^^^^^^^^^^^^ duplicate definition
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

